### PR TITLE
support aten::type_as in the pytorch frontend

### PR DESCRIPTION
**Pull request by randxie**
I was trying to convert SSD model in mmdetection to Relay, and find the type_as is not supported. This PR adds support for aten::type_as operator.

@masahi

### CODE REVIEW

File: tests/python/frontend/pytorch/test_forward.py [hunk 1]
```diff
@@ -836,6 +836,20 @@ def forward(self, *args):
...
 
+
+def test_type_as():
+    torch.set_grad_enabled(False)
+    input_shape = [1, 3]
+
+    class TypeAsInt32(Module):
+        def forward(self, *args):
+            int32_tensor = torch.zeros(1, 3, dtype=torch.int32)
+            return args[0].type_as(int32_tensor)
```
Comment (t-vi) [inline tests/python/frontend/pytorch/test_forward.py:847]: Great stuff!

I would suggest to test more dtypes.
(One of my main goals with #5779 is to be able to use input_types safely, and I hit troubles with it before.)

Comment (randxie) [reply tests/python/frontend/pytorch/test_forward.py]: Done. Did not realize the torch types and tvm type has different naming. Updated the test with more types covered.

**Comment by masahi** [comment]
Thanks @randxie @t-vi

**Comment by junrushao1994** [comment]
Hey the master CI failure seems relevant to this PR. Could you take a look? Thanks!

**Comment by tqchen** [comment]
Followup https://github.com/apache/incubator-tvm/pull/5799

**Pull request merged**